launch_test.go

```diff
@@ -958,6 +986,74 @@ func TestFetchParentBuildMetaWriteError(t *testing.T) {
 	}
 }
 
+func TestFetchParentBuildsMeta(t *testing.T) {
+	actual := make(map[string]interface{})
+	TestParentBuildIDs := []float64{1111, 2222}
+	IDs := make([]interface{}, len(TestParentBuildIDs))
+	for i, s := range TestParentBuildIDs {
+		IDs[i] = s
+	}
+	TestMeta1 := map[string]interface{}{
+		"foo":    "bar",
+		"batman": "robin",
+	}
+	TestMeta2 := map[string]interface{}{
+		"foo":    "baz",
+		"wonder": "woman",
+	}
+
+	oldWriteFile := writeFile
+	defer func() { writeFile = oldWriteFile }()
+
+	api := mockAPI(t, TestBuildID, TestJobID, 0, "RUNNING")
+	api.buildFromID = func(buildID int) (screwdriver.Build, error) {
+		if buildID == TestParentBuildID {
+			return screwdriver.Build(FakeBuild{ID: TestBuildID, Meta: TestMeta1}), nil
+		}
+		if buildID == 2222 {
+			return screwdriver.Build(FakeBuild{ID: TestBuildID, Meta: TestMeta2}), nil
+		}
+		return screwdriver.Build(FakeBuild{ID: TestBuildID, JobID: TestJobID, ParentBuildID: IDs}), nil
+	}
+	marshal = func(v interface{}) (result []byte, err error) {
+		actual = v.(map[string]interface{})
+		return nil, fmt.Errorf("Testing parsing parent event meta")
+	}
+
+	_ = launch(screwdriver.API(api), TestBuildID, TestWorkspace, TestEmitter, TestMetaSpace, TestStoreURL, TestShellBin)
+
+	if actual["foo"] != "baz" {
+		t.Errorf("Error is wrong, got '%v', expected '%v'", actual["foo"], "baz")
+	}
+	if actual["batman"] != "robin" {
+		t.Errorf("Error is wrong, got '%v', expected '%v'", actual["batman"], "robin")
+	}
+	if actual["wonder"] != "woman" {
+		t.Errorf("Error is wrong, got '%v', expected '%v'", actual["wonder"], "woman")
+	}
+}
+
+func TestMergedMetaFromParentBuildsIDs(t *testing.T) {
```

Comment:
what is this test case trying to test? should be covered by `TestFetchParentBuildsMeta ` already, right?

Reply:
Removed